.github/workflows/tests.yml

```diff
@@ -0,0 +1,92 @@
+name: Tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+permissions:
+  contents: read
+  pull-requests: read
+
+jobs:
+
+  test-linux:
+    name: Test on Linux
+    runs-on: ubuntu-latest
+    env:
+      GO_VERSION: oldstable
+    steps:
+
+      - name: Setup /etc/hosts
+        run: |
+          echo "127.0.0.1 example.letsencrypt.org" | sudo tee -a /etc/hosts
+          echo "127.0.0.1 elpmaxe.letsencrypt.org" | sudo tee -a /etc/hosts
+
+      # https://github.com/marketplace/actions/checkout
+      - name: Check out code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      # https://github.com/marketplace/actions/setup-go-environment
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - name: apt install
+        run: sudo apt-get install snapd python3-acme python3-josepy
+
+      - name: snap install
+        run: sudo snap install core && sudo snap refresh core
+
+      - name: snap install certbot
+        run: sudo snap install --classic certbot && sudo ln -s /snap/bin/certbot /usr/bin/certbot
+
+      - name: go install tools
+        run: go install golang.org/x/tools/cmd/cover@latest
+
+      - name: go install goveralls
+        run: go install github.com/mattn/goveralls@latest
+
+      # Clone Boulder because the go.mod contains `exclude` directives,
+      # and so the load-generator cannot be installed with a simple `go install`
+      - name: Checkout Boulder repository (load-generator)
+        uses: actions/checkout@v4
+
+        with:
+          repository: letsencrypt/boulder
+          path: boulder
+          ref: release-2024-02-20
+
+      - name: install load-generator
+        run: |
+          cd $GITHUB_WORKSPACE/boulder/test/load-generator
+          go install
+          cd -
```

Comment:
You can use [steps.working-directory](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsworking-directory) to set the pwd instead of using cd.